### DISCUSSION TITLE
Not read the whole

### DISCUSSION THREAD

**Discussion by mengwy13** [zeux/pugixml#512] [General]
Hello
I use pugixml generate a file. 
Like this:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Config description="assign">
	<Vectors>
		<Vector ID="0" name="M" />
		<Vector ID="1" name="FJ" />
		<Vector ID="2" name="HD" />
		<Vector ID="3" name="BAD" />
		<Vector ID="4" name="BBD" />
		<Vector ID="5" name="B" />
		<Vector ID="6" name="FC" />
	</Vectors>
</Config>
```

But I can't read the complete content  using the following code ;
`
	pugi::xml_document doc;
	pugi::xml_parse_result res = doc.load_file(fullName.toStdString().c_str());
	if (res.status!=pugi::status_ok)
	{
		qDebug() << "error" <<endl;
		return;
	}
`
the _buffer in doc is 

```
<?xml version="1.0" encoding="UTF-8"?>
<Config
```

I change some xml file,but the same matter;
Sorry for my poor English;
Thanks ;

**Comment by zeux** [comment]
> the _buffer in doc is

`_buffer` is a private variable; you should not be paying attention to that, the fact that it stops at `Config` is not a problem since that's just how pugixml internally parses data (by adding extra NUL characters to the buffer that the debugger assumes end the string).
